### PR TITLE
ci: Avoid building unnecessary projects for generating packages

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

### <samp>🤖 Generated by Copilot at 80c9564</samp>

This pull request updates the solution filter files for the skia, wasm, and netcoremobile packages to improve the build and test performance and accuracy. It also adds a new solution filter file for the netcoremobile packages and uses it in the `Uno.UI.Build.csproj` file.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: build/filters/Uno.UI-packages-wasm.slnf
Comment (Youssef1313) [inline build/filters/Uno.UI-packages-wasm.slnf:1]: These changes have three savings

1. It should be a faster build, then we will no longer build projects that are not needed.
2. Artifact size is now smaller, which will cause:
    - a faster upload [here](https://github.com/unoplatform/uno/blob/7d0c72e9fd91103ad2a3c358b2a2cdaad72f3984/build/ci/.azure-devops-package-wasm.yml#L70-L76)
    - a faster download [here](https://github.com/unoplatform/uno/blob/7d0c72e9fd91103ad2a3c358b2a2cdaad72f3984/build/ci/.azure-devops-package.yml#L47-L52)


Checking `wasm-bin-UWP.zip` from CI, the removed projects from this slnf are responsible for 428MB out of 1.15GB total (the mentioned sizes are the unzipped sizes). A similar saving would be for WinUI (but I haven't checked the exact sizes for WinUI build)

**Comment by Youssef1313** [comment]
Taking `.NET Mobile Package Binaries UWP_NET8` as example:

Before PR:

- `Building package binaries` step took 7 to 8 minutes
- `CopyFiles` took around 45 sec
- `ArchiveFiles` took around 24 sec
- `PublishBuildArtifacts` took 2 to 3 minutes

With PR:

- `Building package binaries` step took 4 minutes and 12 seconds
- `CopyFiles` took 18 sec
- `ArchiveFiles` took 13 sec
- `PublishBuildArtifacts` took 52 seconds

The whole job is experiencing about 7 minutes savings.

**Comment by Youssef1313** [comment]
Example before PR:

image

PR:

image

**Comment by Youssef1313** [comment]
This PR:

image

Another random PR:

image

---

This PR:

image

Another random PR:

image

---

Clarifications:

- For Pack NuGet:
    - `Download build artifact NugetPackages-Artifacts-netcoremobile` dropped from 2m 36sec to 1m 4sec
    - `Download build artifact NugetPackages-Artifacts-skia` dropped from 1m 39sec to 23sec
    - `Restore binaries structure` dropped from 3m to 1m 21sec

- For Build Binaries job (taking as example `.NET Mobile Package Binaries UWP_NET8`):
    - `Building package binaries` dropped from 7m 45sec to 4m 58sec
    - `CopyFiles` dropped from 45sec to 19sec
    - `ArchiveFiles` dropped from 24sec to 14sec
    - `PublishBuildArtifacts` dropped from 2m 12sec to 56sec

**Review by MartinZikmund — APPROVED** [review]

**Review by ajpinedam — APPROVED** [review]

**Comment by jeromelaban** [comment]
@Youssef1313 doing such a change implies that all samples apps are built by other targets (which is currently the case for net7, but not for net8), but it will need to be carefully updated in the future. This may not be a problem as we move to test all heads properly.

**Comment by Youssef1313** [comment]
@jeromelaban I think they are indeed built by other jobs

**Comment by jeromelaban** [comment]
> Do we care about that?

Not at this time. We'll check again when we move to net8.

**Pull request merged**